Userland/Libraries/LibWeb/HTML/Window.cpp

```diff
@@ -1395,4 +1400,166 @@ JS::NonnullGCPtr<CustomElementRegistry> Window::custom_elements()
+    for (auto& entry : child_navigable_property_set)
+        property_names.set(entry.key, AK::HashSetExistingEntryBehavior::Keep);
+
+    // - the value of the name content attribute for all embed, form, img, and object elements
+    //   that have a non-empty name content attribute and are in a document tree with window's associated Document as their root; and
+    // - the value of the id content attribute for all HTML elements that have a non-empty id content attribute
+    //   and are in a document tree with window's associated Document as their root.
+    const_cast<Window&>(*this).associated_document().for_each_in_subtree([&property_names, this](auto& node) -> IterationDecision {
+        if (is<HTMLElement>(node)) {
+            auto& element = static_cast<HTMLElement&>(node);
+            if (&element.document() != &associated_document())
+                return IterationDecision::Continue;
+            if (is<HTMLEmbedElement>(element) || is<HTMLFormElement>(element) || is<HTMLImageElement>(element) || is<HTMLObjectElement>(element)) {
+                if (auto const& name = element.attribute(AttributeNames::name); name.has_value())
+                    property_names.set(name.value(), AK::HashSetExistingEntryBehavior::Keep);
+            }
+            if (auto const& name = element.attribute(AttributeNames::id); name.has_value())
+                property_names.set(name.value(), AK::HashSetExistingEntryBehavior::Keep);
+        }
+        return IterationDecision::Continue;
+    });
+
+    // FIXME: Many copies here. Would be nice to be able to return Vector<String>
+    Vector<DeprecatedString> names;
+    names.ensure_capacity(property_names.size());
+    for (auto const& name : property_names) {
+        names.append(name.to_deprecated_string());
+    }
+    return names;
+}
+
+// https://html.spec.whatwg.org/#named-access-on-the-window-object
+WebIDL::ExceptionOr<JS::Value> Window::named_item_value(DeprecatedFlyString const& name) const
+{
+    // To determine the value of a named property name in a Window object window, the user agent must return the value obtained using the following steps:
+
+    // 1. Let objects be the list of named objects of window with the name name.
+    // NOTE: There will be at least one such object, since the algorithm would otherwise not have been invoked by Web IDL.
+    auto objects = const_cast<Window&>(*this).named_objects(name);
+
+    // 2. If objects contains a navigable, then:
+    if (!objects.navigables.is_empty()) {
+        // 1. Let container be the first navigable container in window's associated Document's descendants whose content navigable is in objects.
+        JS::GCPtr<NavigableContainer> container = nullptr;
+        const_cast<Window&>(*this).associated_document().for_each_in_subtree_of_type<HTML::NavigableContainer>([&](HTML::NavigableContainer& navigable_container) {
```

Comment:
the const correctness of these things is all screwed up, I'm not quite sure how to fix it.

Reply:
:yakshrug:

Reply:
Maybe `named_item_value()` being `const` is wishful thinking.